src/GoogleFitOnFhir/Services/GoogleFitDataImporter.cs

```diff
@@ -98,34 +96,67 @@ public async Task Import(string userId, CancellationToken cancellationToken)
             var endDate = endDateDto.ToUnixTimeMilliseconds() * 1000000;
             var datasetId = startDate + "-" + endDate;
 
-            // Get dataset for each dataSource
+            List<Task> datasetRequestTasks = new List<Task>();
+
+            // Add a task to get the Dataset for each dataSource
             foreach (var datasourceId in dataSourcesList.DatasourceIds)
```

Comment:
After seeing how quickly my list of data sources is growing, I actualy think that we need to put some constraints on the number of parallel tasks that can run at one time or we could potentially run into CPU or mempry issues.

We actually have a [ParallelTaskWorker class](https://github.com/microsoft/iomt-fhir/blob/main/src/lib/Microsoft.Health.Common/Service/ParallelTaskWorker.cs) that is in the Microsoft.Health.Common package that handles some of the basic things we would want to manage.

[Here](https://github.com/microsoft/iomt-fhir/blob/726cf7dd7afd011b9e593cb6eae425d79ef7ab56/src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementFhirImportService.cs#L61) is an example implementation.

It makes sense to me that this class could inherit from the ParallelTaskWorker class, and it would allow us to configure the number of parallel tasks and also filter exceptions if needed.